### ISSUE TITLE
Improving SolverEngine

### ISSUE THREAD

**Issue by rainoftime** [JonathanSalwan/Triton#618]
In `src/libtriton/engines/solver/solverEngine.cpp`

~~~~
132        /* First, set the QF_AUFBV flag  */
133        formula << "(set-logic QF_BV)";
~~~~

If the logic is QF_AUFBV(quantifier-free bit-vector and array),  I thank line 13 should be `(set-logic QF_AUFBV)`

**Comment by rainoftime** [comment]
~~~~
        /* Then, delcare all symbolic variables */
        formula << this->symbolicEngine->getVariablesDeclaration();
        /* And concat the user expression */
        formula << this->symbolicEngine->getFullAst(node);
        /* Create the context and AST */
        Z3_ast ast = Z3_parse_smtlib2_string(ctx, formula.str().c_str(), 0, 0, 0, 0, 0, 0);
        z3::expr eq(ctx, ast);
        /* Create a solver and add the expression */
        solver.add(eq);
~~~~

It seems currently we must output the `contents` of a query into a string, and then calls z3 to parse the string. Can Triton use z3 C++ API to interact with the solver?

**Comment by JonathanSalwan** [comment]
Already fixed on the `dev-v0.6` branch.
